### PR TITLE
fix(Splitter): fix the issue about Splitter had unexpected gaps in Flex

### LINKED ISSUE #51095: There are unexpected gaps when the Splitter is in Flex
### Reproduction link

[https://ant.design/~demos/splitter-demo-control](https://ant.design/~demos/splitter-demo-control)

https://github.com/user-attachments/assets/eba3d497-1159-4f33-870b-cc7f73efd817

### Steps to reproduce

1. Press the left mouse button at the splitBar to reproduce

### What is expected?

Do not add unexpected gap in Flex Component

### What is actually happening?

Display unexpected gap in Flex Component

| Environment | Info |
| --- | --- |
| antd | 5.21.2 |
| React | 18.3.1 |
| System | macOS 14.6.1 (23G93) |
| Browser | Chrome 版本 129.0.6668.59（正式版本） (x86_64) |

### PR DESCRIPTION
**Pull request by kiner-tang**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [ ] 🆕 New feature
- [X] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues

> - Describe the source of related requirements, such as links to relevant issue discussions.
> - For example: close #xxxx, fix #xxxx

close #51095 

### 💡 Background and Solution

> - The specific problem to be addressed.
> - List the final API implementation and usage if needed.
> - If there are UI/interaction changes, consider providing screenshots or GIFs.

### 📝 Change Log

> - Read [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) like a cat tracks a laser pointer.
> - Describe the impact of the changes on developers, not the solution approach.
> - Reference: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Fix the issue about Splitter had unexpected gaps in Flex |
| 🇨🇳 Chinese | 解决 Splitter 组件在 Flex 组件下时出现异常间距的问题 |

### CODE REVIEW

File: components/splitter/Splitter.tsx [hunk 1]
```diff
@@ -146,65 +146,62 @@ const Splitter: React.FC<React.PropsWithChildren<SplitterProps>> = (props) => {
...
+          return (
+            <React.Fragment key={`split-panel-${idx}`}>
+              {panel}
+              {splitBar}
+            </React.Fragment>
+          );
+        })}
+        {/* Fake mask for cursor */}
+        {typeof movingIndex === 'number' && (
+          <div aria-hidden className={classNames(maskCls, `${maskCls}-${layout}`)} />
```
Comment (li-jia-nan) [inline components/splitter/Splitter.tsx:201]: 是把 mask 移动到 ResizeObserver 里面了吗？

Comment (kiner-tang) [reply components/splitter/Splitter.tsx:201]: 是的，不然如果 Splitter 在 Flex 布局下面的话，mask 元素也算是一个 FlexItem，会导致出现异常的 gap

Comment (kiner-tang) [reply components/splitter/Splitter.tsx:201]: 对的，
image

Comment (li-jia-nan) [reply components/splitter/Splitter.tsx:201]: 我感觉没啥问题，让大佬们再看下

**Review by li-jia-nan — APPROVED** [review]

**Review by afc163 — APPROVED** [review]

**Pull request merged**